src/molecular-weight.jl

```diff
@@ -0,0 +1,105 @@
+
+# Creating the array AA_WEIGHTS of length 28 to list all weights. In ambigous cases, let's list the string "Undefined", so if called, it returns an error
+AA_WEIGHTS = [89.09, 174.20, 132.12, 133.10, 121.16, 146.14, 147.13, 75.07, 155.15, 131.17, 131.17, 146.19, 149.21, 165.19, 115.13, 105.09, 119.12, 204.22, 181.19, 117.15, 255.31, 168.06, "Undefined", 131.17, "Undefined", "Undefined", 0, 0]
+
+# Defining the function molecular_weight, which accepts an amino acid sequence and return molecular weight in  g/mol. Note that the function also account for lost of water for each peptide bond formation 
+function molecular_weight(aa_seq::AASeq)
+    weight = 0
+    for aa in aa_seq
+        aa_weight = AA_WEIGHTS[reinterpret(UInt8, aa) + 1]
+        weight += aa_weight
+    end
+    return round(weight - ((length(aa_seq) - 1) * 18.02); digits = 3)
+end
+
+# Sources of the molecular weights of each defined nucleotide
+# DNA Adenine                 dAMP ->  Source = https://pubchem.ncbi.nlm.nih.gov/compound/12599
+# RNA Adenine                 AMP  ->  Source = https://pubchem.ncbi.nlm.nih.gov/compound/6083
+# DNA Cytosine                dCMP ->  Source = https://pubchem.ncbi.nlm.nih.gov/compound/13945
+# RNA Cytosine                CMP  ->  Source = https://pubchem.ncbi.nlm.nih.gov/compound/6131  
+# DNA Guanine                 dGMP ->  Source = https://pubchem.ncbi.nlm.nih.gov/compound/1135398597
+# RNA Guanine                 GMP  ->  Source = https://pubchem.ncbi.nlm.nih.gov/compound/135398631
+# DNA Thymine                 dTMP ->  Source = https://pubchem.ncbi.nlm.nih.gov/compound/9700
+# Calculations following the following guidelines: https://ymc.eu/files/imported/publications/556/documents/YMC-Expert-Tip---How-to-calculate-the-MW-of-nucleic-acids.pdf
+
+# Creating the arrays DNA_WEIGHTS and RNA_WEIGHTS to list all weights. In ambigous cases, let's list the string "Undefined", so if called, it returns an error
+# The values already account for loss of water = - 18.02
+DNA_WEIGHTS = [0, 313.2, 289.18, "Undefined", 329.2, "Undefined", "Undefined", "Undefined", 304.19, "Undefined", "Undefined", "Undefined", "Undefined", "Undefined", "Undefined", "Undefined"]
+
+RNA_WEIGHTS = [0, 329.2, 305.18, "Undefined", 345.2, "Undefined", "Undefined", "Undefined", 306.16, "Undefined", "Undefined", "Undefined", "Undefined", "Undefined", "Undefined", "Undefined"]
+
+# Defining a function that accepts nucleotide sequences (either DNA or RNA) and calculate the molecular weight in g/mol
+# Note the additional keywords alphabet, five_terminal_state and strand_number, which have default cases, if not specified
+# In case of using double strand, the five_terminal_state of the complmentary strand is assumed to be the same as the input strand
+# Function only accepts double as strand_number for DNA sequences
+function molecular_weight(nucseq::NucSeq, alphabet=:DNA, five_terminal_state=:hydroxyl, strand_number=:single)
```

Comment:
Here, do not have an argument `alphabet` - instead, have distinct methods for `BioSequence{<:DNAAlphabet}` and `BioSequence{<:RNAAphabet}`. These two methods only need to load the appropriate array, and can then call the same method.